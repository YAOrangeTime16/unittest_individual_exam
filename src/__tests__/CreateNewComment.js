import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CreateNewComment from '../components/CreateNewComment';

describe('form', ()=>{
    const component = <CreateNewComment 
    postId=''
    author=''
    updateComments={jest.fn()} />;

    it('should renders the component', ()=>{
        const wrapper = render(component);
        expect(wrapper).toBeTruthy();
    });

    it('should change the state', ()=>{
        const wrapper = shallow(component);
        const event = {target: {name: 'comment', value: 'test value'}}
        wrapper.find('textarea').simulate('change', event);
        expect(wrapper.state('comment')).toBe('test value');
    });

    it('should re-set the state to empty', ()=>{
        const wrapper = shallow(component);
        wrapper.setState({comment: 'hi'});
        const mockEvent = { preventDefault: jest.fn() };
        wrapper.find('form').simulate('submit', mockEvent);
        expect(wrapper.state('comment')).toHaveLength(0);
    });
    
});