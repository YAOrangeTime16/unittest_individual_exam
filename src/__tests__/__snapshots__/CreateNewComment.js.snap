// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form should match its snapshot 1`] = `
<form
  class="container mx-auto flex flex-col p-6"
>
  <label
    class="mb-2"
    for="comment"
  >
    Comment
  </label>
  <textarea
    class="shadow rounded border mb-2 p-2"
    id="comment"
    name="comment"
  >
    
  </textarea>
  <input
    class="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
    type="submit"
    value="Comment"
  >
    
  </input>
</form>
`;
